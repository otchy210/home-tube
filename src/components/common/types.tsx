export type HTMLElementProps = React.HTMLAttributes<HTMLElement>;

export type HTMLDivProps = React.HTMLAttributes<HTMLDivElement>;

export type HTMLButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement>;

export type HTMLFormProps = React.FormHTMLAttributes<HTMLFormElement>;

export type HTMLLabelProps = React.LabelHTMLAttributes<HTMLLabelElement>;

export type HTMLInputProps = Omit<React.InputHTMLAttributes<HTMLInputElement>, 'type'> & {
    isInvalid?: boolean;
};

export type HTMLCheckboxProps = HTMLInputProps & {
    label?: string;
};

export type HTMLSelectProps = React.SelectHTMLAttributes<HTMLSelectElement>;
