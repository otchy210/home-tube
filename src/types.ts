export type VideoViewMode = 'normal' | 'theater' | 'fullScreen';

export interface FolderData {
    path: string;
    name: string;
    folders: FolderData[];
}

export interface TreeNode {
    key: string;
    title: string;
    children?: TreeNode[];
}
