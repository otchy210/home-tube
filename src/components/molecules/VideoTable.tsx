import { VideoValues } from '@otchy/home-tube-api/dist/types';
import React from 'react';
import { Col, Row } from 'react-bootstrap';
import VideoCard from './VideoCard';

type Props = {
    videos: VideoValues[];
};

const VideoTable: React.FC<Props> = ({ videos }: Props) => {
    return (
        <Row>
            {videos.map((video) => {
                return (
                    <Col xs={12} md={6} lg={4} xxl={3} className="px-1" key={`video-card-${video.key}`}>
                        <VideoCard video={video} />
                    </Col>
                );
            })}
        </Row>
    );
};

export default VideoTable;
