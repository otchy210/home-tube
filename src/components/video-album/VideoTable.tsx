import { VideoValues } from '@otchy/home-tube-api/dist/types';
import React from 'react';
import { Col, Row } from '../common/layouts';
import VideoCard from './VideoCard';

type Props = {
    videos: VideoValues[];
};

const VideoTable: React.FC<Props> = ({ videos }: Props) => {
    return (
        <Row>
            {videos.map((video) => {
                return (
                    <Col width={[12, 12, 6, 4, 4, 3]} className="px-1" key={`video-card-${video.key}`}>
                        <VideoCard video={video} />
                    </Col>
                );
            })}
        </Row>
    );
};

export default VideoTable;
