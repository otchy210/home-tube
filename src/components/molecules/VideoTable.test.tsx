import React from 'react';
import { render } from '@testing-library/react';
import VideoTable from './VideoTable';
import { VideoDocument } from '@otchy/home-tube-api/dist/types';
import { BrowserRouter } from 'react-router-dom';

describe('VideoTable', () => {
    it('renders properly', () => {
        const videos = new Array(6).fill('').map((_, index) => {
            return {
                id: index + 1,
                values: {
                    name: `Video-${index + 1}`,
                },
            };
        }) as VideoDocument[];
        const { container } = render(
            <BrowserRouter>
                <VideoTable videos={videos} />
            </BrowserRouter>
        );
        expect(container).toMatchSnapshot();
    });
});
