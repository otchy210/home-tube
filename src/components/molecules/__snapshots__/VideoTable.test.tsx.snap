// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoTable renders properly 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="px-1 col-xxl-3 col-lg-4 col-md-6 col-12"
    >
      <div
        class="mt-4 card"
      >
        <a
          href="/view?key=key-1"
          style="text-decoration: none;"
        >
          <div
            class="sc-bdvvtL iRLeeU card-img-top"
          >
            <div
              class="sc-gsDKAQ esGSdz"
            />
            <img
              class="sc-dkPtRN cQjecG"
              src="http://dummy-api-host/snapshot?key=key-1"
              style="opacity: 1;"
            />
            <div
              class="sc-hKwDye ihnUmD hstack m-1"
            />
          </div>
        </a>
        <div
          class="card-body"
        >
          <div
            class="fs-6 text-truncate card-title h5"
            title="Video-1"
          >
            <a
              href="/view?key=key-1"
              style="text-decoration: none;"
            >
              Video-1
            </a>
          </div>
          <div
            class="text-muted card-subtitle h6"
          >
            <div
              class="sc-iCfMLu iCHdgv"
            >
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="px-1 col-xxl-3 col-lg-4 col-md-6 col-12"
    >
      <div
        class="mt-4 card"
      >
        <a
          href="/view?key=key-2"
          style="text-decoration: none;"
        >
          <div
            class="sc-bdvvtL iRLeeU card-img-top"
          >
            <div
              class="sc-gsDKAQ esGSdz"
            />
            <img
              class="sc-dkPtRN cQjecG"
              src="http://dummy-api-host/snapshot?key=key-2"
              style="opacity: 1;"
            />
            <div
              class="sc-hKwDye ihnUmD hstack m-1"
            />
          </div>
        </a>
        <div
          class="card-body"
        >
          <div
            class="fs-6 text-truncate card-title h5"
            title="Video-2"
          >
            <a
              href="/view?key=key-2"
              style="text-decoration: none;"
            >
              Video-2
            </a>
          </div>
          <div
            class="text-muted card-subtitle h6"
          >
            <div
              class="sc-iCfMLu iCHdgv"
            >
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="px-1 col-xxl-3 col-lg-4 col-md-6 col-12"
    >
      <div
        class="mt-4 card"
      >
        <a
          href="/view?key=key-3"
          style="text-decoration: none;"
        >
          <div
            class="sc-bdvvtL iRLeeU card-img-top"
          >
            <div
              class="sc-gsDKAQ esGSdz"
            />
            <img
              class="sc-dkPtRN cQjecG"
              src="http://dummy-api-host/snapshot?key=key-3"
              style="opacity: 1;"
            />
            <div
              class="sc-hKwDye ihnUmD hstack m-1"
            />
          </div>
        </a>
        <div
          class="card-body"
        >
          <div
            class="fs-6 text-truncate card-title h5"
            title="Video-3"
          >
            <a
              href="/view?key=key-3"
              style="text-decoration: none;"
            >
              Video-3
            </a>
          </div>
          <div
            class="text-muted card-subtitle h6"
          >
            <div
              class="sc-iCfMLu iCHdgv"
            >
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="px-1 col-xxl-3 col-lg-4 col-md-6 col-12"
    >
      <div
        class="mt-4 card"
      >
        <a
          href="/view?key=key-4"
          style="text-decoration: none;"
        >
          <div
            class="sc-bdvvtL iRLeeU card-img-top"
          >
            <div
              class="sc-gsDKAQ esGSdz"
            />
            <img
              class="sc-dkPtRN cQjecG"
              src="http://dummy-api-host/snapshot?key=key-4"
              style="opacity: 1;"
            />
            <div
              class="sc-hKwDye ihnUmD hstack m-1"
            />
          </div>
        </a>
        <div
          class="card-body"
        >
          <div
            class="fs-6 text-truncate card-title h5"
            title="Video-4"
          >
            <a
              href="/view?key=key-4"
              style="text-decoration: none;"
            >
              Video-4
            </a>
          </div>
          <div
            class="text-muted card-subtitle h6"
          >
            <div
              class="sc-iCfMLu iCHdgv"
            >
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="px-1 col-xxl-3 col-lg-4 col-md-6 col-12"
    >
      <div
        class="mt-4 card"
      >
        <a
          href="/view?key=key-5"
          style="text-decoration: none;"
        >
          <div
            class="sc-bdvvtL iRLeeU card-img-top"
          >
            <div
              class="sc-gsDKAQ esGSdz"
            />
            <img
              class="sc-dkPtRN cQjecG"
              src="http://dummy-api-host/snapshot?key=key-5"
              style="opacity: 1;"
            />
            <div
              class="sc-hKwDye ihnUmD hstack m-1"
            />
          </div>
        </a>
        <div
          class="card-body"
        >
          <div
            class="fs-6 text-truncate card-title h5"
            title="Video-5"
          >
            <a
              href="/view?key=key-5"
              style="text-decoration: none;"
            >
              Video-5
            </a>
          </div>
          <div
            class="text-muted card-subtitle h6"
          >
            <div
              class="sc-iCfMLu iCHdgv"
            >
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="px-1 col-xxl-3 col-lg-4 col-md-6 col-12"
    >
      <div
        class="mt-4 card"
      >
        <a
          href="/view?key=key-6"
          style="text-decoration: none;"
        >
          <div
            class="sc-bdvvtL iRLeeU card-img-top"
          >
            <div
              class="sc-gsDKAQ esGSdz"
            />
            <img
              class="sc-dkPtRN cQjecG"
              src="http://dummy-api-host/snapshot?key=key-6"
              style="opacity: 1;"
            />
            <div
              class="sc-hKwDye ihnUmD hstack m-1"
            />
          </div>
        </a>
        <div
          class="card-body"
        >
          <div
            class="fs-6 text-truncate card-title h5"
            title="Video-6"
          >
            <a
              href="/view?key=key-6"
              style="text-decoration: none;"
            >
              Video-6
            </a>
          </div>
          <div
            class="text-muted card-subtitle h6"
          >
            <div
              class="sc-iCfMLu iCHdgv"
            >
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
              <div
                class="sc-gKclnd cZoYU"
              >
                [MockedStarVoidSvg width="16" height="16" /]
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
